pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -504,6 +505,10 @@ def _format_checkpoint_name(
             for group in groups:
                 name = group[1:]
 
+                if (name == 'monitor' and monitor is not None):
+                    metrics['monitor'] = metrics[monitor]
+                    name = monitor
+
```

Comment:
@tchaton  @Lucklyric I also prefer @carmocca 's suggestion here: https://github.com/PyTorchLightning/pytorch-lightning/issues/7260#issuecomment-836619959

this keeps the callback simpler and avoids special casing like this. wdyt?

Reply:
I agree with it